### DISCUSSION TITLE
I got an 'unofficial' G (Giant) and it is a noticeable improvement over L (Large)

### DISCUSSION THREAD

**Discussion by gituser123456789000** [nagadomi/nunif#184] [General]
I got an 'unofficial' G (Giant) and it is a noticeable improvement over L (Large)

We also have a new Tiling program that's pulling out good details. I wouldn't say completely accurate, due to the issues involved with tiling, but the maps can be converted at low strength and look pretty good as far as fine details, like noses sticking out instead of a flat face for example

Here's an example of S vs B vs L vs unofficial G
![owl og 1s 16bit_low](https://github.com/user-attachments/assets/2e130a9f-611a-4eb4-8d67-0f1ac9203fb7)
![owl og 2b 16bit_low](https://github.com/user-attachments/assets/93370fb8-ae2e-41fa-acff-e4bcf8b2382e)
![owl og 3l 16bit_low](https://github.com/user-attachments/assets/4e72c8e0-d874-4808-a4ab-699418b2d4c6)
![owl og 4g 16bit_low](https://github.com/user-attachments/assets/350f5f31-d926-4188-a5d3-5bb44a4ea7de)

Here's an example of 4,4 Tiled versions of S vs B vs L vs unofficial G
![owl og 1s tiled_depth_(4, 4)](https://github.com/user-attachments/assets/9b31fdb4-dab1-4030-ad02-a5abbf44faac)
![owl og 2b tiled_depth_(4, 4)](https://github.com/user-attachments/assets/9bae5847-2b07-4428-8da7-d8673ed1eefc)
![owl og 3l tiled_depth_(4, 4)](https://github.com/user-attachments/assets/97b0da6b-4038-4655-9787-66fc4782fddf)
![owl og 4g tiled_depth_(4, 4)](https://github.com/user-attachments/assets/b141d9da-e209-42ab-b9fd-74159adfd837)

**Comment by gituser123456789000** [comment]
Please add the code for Any_V2_G to function in iw3

**Comment by gituser123456789000** [comment]
I got Giant running in iw3 pretty quickly... surprisingly (because I'm not a coder).

Here's an example of v2L vs v2 Giant (unofficial) running through iw3 (the Giant result actually looks better than the alternate program I was previously able to run unofficial Giant in)

![owl og 3l](https://github.com/user-attachments/assets/ed964113-cec9-4b98-b20f-71abca468751)
![owl og 4g](https://github.com/user-attachments/assets/5de3aaf7-a7cc-44d7-bcee-d825341f45cf)

and here are the file edits I made... I assume I got everything, since the output seems to be as it should be.. but I may have missed something or done too much.. like I may have added stuff for Any v1 G that doesn't exists..

<img width="752" alt="iw3edit1" src="https://github.com/user-attachments/assets/84b5cdf7-2c1a-4b75-86b0-7bd834a0cdc5">
<img width="798" alt="iw3edit2" src="https://github.com/user-attachments/assets/3a0fff6d-e1ad-4c06-84e3-071f5fad88fe">
<img width="738" alt="iw3edit3" src="https://github.com/user-attachments/assets/707cf139-07c3-48c7-a226-e1d7291c0f92">
<img width="768" alt="iw3edit4" src="https://github.com/user-attachments/assets/18ad33c1-d654-4d9c-95b2-9dda6954fbd3">
<img width="948" alt="iw3edit5" src="https://github.com/user-attachments/assets/ee17474b-80b1-4e80-99d4-42cc3906e3ce">
<img width="549" alt="iw3edit6" src="https://github.com/user-attachments/assets/277dae91-87f1-46d4-bd1c-7024aec64206">
<img width="540" alt="iw3edit7" src="https://github.com/user-attachments/assets/150e23c6-88eb-4512-abe9-5710579b7a42">
<img width="907" alt="iw3edit8" src="https://github.com/user-attachments/assets/7098597a-039f-4902-a4e7-b3867e65f1b4">
<img width="722" alt="iw3edit9" src="https://github.com/user-attachments/assets/1709be45-df6e-4f78-9ba2-fd78720b592f">
<img width="722" alt="iw3edit10" src="https://github.com/user-attachments/assets/5c37864b-d7ae-406d-b8f9-d8cc69e4d377">
<img width="775" alt="iw3edit11" src="https://github.com/user-attachments/assets/de50eeb6-861f-4a6e-83a2-a6049e9ef25e">
<img width="742" alt="iw3edit12" src="https://github.com/user-attachments/assets/3fb00132-757b-41cf-b3bd-3624ef8978f8">
<img width="1002" alt="iw3edit13" src="https://github.com/user-attachments/assets/62435a6b-6c9f-49b8-a301-f84af502bdd4">
<img width="933" alt="iw3edit14" src="https://github.com/user-attachments/assets/732f489e-48ad-4920-85ec-9715897f39b9">

**Reply by diverswan9** [reply]
It does work.

**Reply by carlobs** [reply]
How can i edit the .pyc files? i can't with standard editors

**Reply by gituser123456789000** [reply]
> How can i edit the .pyc files? i can't with standard editors

I used Notepad++ for all of the code edits

**Comment by gituser123456789000** [comment]
Here's a sample from iw3 showing how much better (unofficial) Giant picked up light sabers compared to Large: 

![the-acolyte-extended-universe-1653481895051 3l](https://github.com/user-attachments/assets/14617ed1-2ce2-4365-8005-83900745768e)
![the-acolyte-extended-universe-1653481895051 4g](https://github.com/user-attachments/assets/b9f65410-20ae-4942-bce3-ff9619f7afc9)
![the-acolyte-extended-universe-1653481895051](https://github.com/user-attachments/assets/9ffe8746-9ead-43fe-918a-b897a05066b1)

**Comment by gituser123456789000** [comment]
A couple more examples comparing Large vs (unofficial) Giant:

![DAK_Pandora_LowAngleNight_DC_v04MerchHD 3l](https://github.com/user-attachments/assets/035273a9-9aab-4667-9e31-deefeabbff35)
![DAK_Pandora_LowAngleNight_DC_v04MerchHD 4g](https://github.com/user-attachments/assets/be4d1d77-a8ac-4331-90f9-c67b5c572a85)
![DAK_Pandora_LowAngleNight_DC_v04MerchHD](https://github.com/user-attachments/assets/a87af282-323e-40b4-b1b6-5bee16dc085b)


![Input p-6zGDVwxRXutNNLg9894YRJz6LPvnY5uB8GBxLB1N0 3l](https://github.com/user-attachments/assets/817846e7-f3c2-4107-8f75-8f38fdcfaf5b)
![Input p-6zGDVwxRXutNNLg9894YRJz6LPvnY5uB8GBxLB1N0 4g](https://github.com/user-attachments/assets/7586da5b-8dc2-48b8-b0c1-03edcee66fd9)
![Input p-6zGDVwxRXutNNLg9894YRJz6LPvnY5uB8GBxLB1N0](https://github.com/user-attachments/assets/2114c6ac-cd98-4b7e-a458-fd0b65833803)


Attention to detail.. you can see Giant improves on what Large is already capable of and catches a bunch of issues that Large has

**Comment by asdf31jsa** [comment]
i think i made all changes like posted here, got no error, but only black images. has something changed since this was posted?

**Reply by gituser123456789000** [reply]
> i think i made all changes like posted here, got no error, but only black images. has something changed since this was posted?

Do you have discord? We can review there. I haven't updated iw3 since making these changes for Giant, so I don't know if files were changed, but I'd assume it's more likely you missed something or have a typo or don't have the model. We can try to figure it out on discord

**Reply by asdf31jsa** [reply]
thanks for your response! it looks like just "fp16" activated doesnt work. it kinda works (but buggy) with: "v2_vitg": [384, [1536, 1536, 1536, 1536], [4, 11, 17, 23]],". but without fp16 activated it also works with 9, 19, 29, 29.

**Reply by gituser123456789000** [reply]
> thanks for your response! it looks like just "fp16" activated doesnt work. it kinda works (but buggy) with: "v2_vitg": [384, [1536, 1536, 1536, 1536], [4, 11, 17, 23]],". but without fp16 activated it also works with 9, 19, 29, 29.

Check my note next to it.. I found the official numbers and check your last number.. 39, not 29 for the last one, double check

9, 19, 29, 39

And you have the Giant model?

**Reply by asdf31jsa** [reply]
yeah that was just a typo in my message ;) without fp16 enabled it works. i have depth_anything_v2_vitg.pth with filesize: 4.914.781

**Reply by gituser123456789000** [reply]
Yeah, that's what I have.. 4,914,781

Test the star wars image above and compare Large vs Giant.. see that your Giant and changes picks up the light sabers and you'll know you should be good

**Reply by francdn** [reply]
Is there anyway to make this work with FP16 enabled?

**Reply by gituser123456789000** [reply]
> Is there anyway to make this work with FP16 enabled?

I just tested and see that FP16 gives an all black result. That's somewhere further in the code that nagadomi would probably have to fix, but they likely won't until Giant releases officially. I don't use FP16 for quality reasons, but I understand some might want it for speed or lower tier GPU usage if it uses less VRAM. I can't help with that. Someone would have to find in the code where this is a problem and I'm not a coder. I just know enough to find and alter relatively easy things. Luckily getting Giant to work as it does was relatively easy. I'll take a look and see if I can find anything with FP16 and why Giant doesn't work with it, but assume I probably won't be able to and we'll have to wait until nagadomi does it officially

**Reply by francdn** [reply]
What difference in quality do you notice between having FP16 and not?

**Reply by gituser123456789000** [reply]
> What difference in quality do you notice between having FP16 and not?

I don't remember what the exact difference was. I tested many months ago, but did find something to be better with FP32 (FP16 unchecked) over FP16. It was likely seen when breaking video down into frames and rapidly comparing back and forth.

One example I remember was I believe the Paramount logo.. stars come in from the side and form a circle around the mountain. Flicker reduction made a big difference there in keeping the mountain much more steady.. otherwise it was shaking a lot without it. And the stars themselves had a roundness to them, but was noticeably improved with TTA On and  FP32 (FP16 unchecked). I don't remember exactly what FP32 did better than FP16, but there was something somewhere or else I wouldn't use it. It could have also cleaned up some artifacts. I'd have to test again to say for sure.

If you're using Windows, the default image viewer likely isn't good for comparing frames. Mine had a flash in between switching images back and forth . It wasn't an instant transition to easy spot differences. I'd recommend Pictureflect Photo Viewer, it's free, or probably has some paid option for extra features. For ripping frames, I use DeepFaceLab, which is a face-swapping program.. but you don't need to do that.. it has scripts for cutting video clips or extracting video frames very easily. You put your test clip in the workspace folder, change the name to data_dst.. then go back to the main DeepFaceLab folder and double click on 3) extract images from video data_dst FULL FPS. This will rip every frame into a folder. You can rename the folder . Add something to the end of the file names in there. Do the process again for a second test clip.. in this case one test with FP16, one with FP32. Then combine the image folders and you can rapidly switch back and forth and instantly see any differences.

To make a clip.. like 1 second, so you're not ripping tons of frames.. you drag and drop your test video onto 3) cut video (drop video on me).bat, and follow the options that pop up in command prompt, asking what times to clip at and how high of bitrate you want. I check the Properties of the test clip beforehand to see what the bitrate is and put in a number a bit above that, so I don't lose quality for when I'm trying to compare

**Reply by francdn** [reply]
Thanks a lot for the detailed explanation. I only did a comparison test with the Large checkpoint and did not notice any differences between 16 and 32 FP apart from the time it took to complete the process. Also, I have not noticed any significant difference between Large and Giant but I have only done one comparison so far. I was hoping that Giant could detect better fine detail like hair but maybe training at 518 depth resolution is too low?

**Reply by gituser123456789000** [reply]
Giant is more accurate and picks up more detail.. easily seen with the lightsabers example above.. and seen in the other examples if you look closer or switch back and forth between them

518 is max and most accurate. You can go higher and it will be more 'detailed' but it won't be correct. It will be less accurate. You'll get depth degradation, where there will be holes in the depth like rust holes in metal, most easily seen near edges.. and inaccuracies in depth, like sunken in neck and chest area and many others depending on the content

Giant is for accuracy, best quality result.

Simple to test.. just run at 518 and like 1036 for still images and compare.. output the depth maps and switch back and forth .. output in 3D and compare back and forth. If you know what you're looking at, you'll likely see more detail in the depth map due to higher resolution, but you're likely to also see depth degradation. Converted to 3D, you'll likely see the inaccuracies of higher than 518 depth resolution with Depth-Anything V2

It's possible that low depth Strength, the higher res could look fine.. but then what's the point of having barely any depth effect?

Could depend on content and the screen size to viewing distance ratio you're viewing from.. but I think on a flat screen with 3D like anaglyph, in-screen 1.0 convergence plane should be best for a simple conversion with 3.0 depth and row flow 3. For VR SBS.. opposite 0.0 convergence plane and 2.0 depth row flow 3 should be better settings, but tweek to your preference. 3 depth for VR 0.0 convergence seemed too much, becoming inaccurate, but could depend on content

**Comment by francdn** [comment]
Is the depth resolution also 518 for the Giant checkpoint?

**Reply by gituser123456789000** [reply]
Should still be 518.. I use 518. Assume it's 518, but if I find anything saying otherwise, I'll let you know.
